### PR TITLE
Add flake8 hook to pre-commit

### PR DESCRIPTION
**Pull request by slsh1o**
Add flake8 hook with isort 5. But at the same time with a lot of `SyntaxError` from unrelated files. Should I just ignore them by adding them to `exclude` list or there is a better way?

### CODE REVIEW

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  # The package has imports exposing private things to the public:
+  src/pylibsshext/__init__.py: WPS412
+
+  # Exclude errors that don't make sense for Cython
+  # Examples:
+  # * "E211 whitespace before '('" happening to "typedef int (*smth) ..."
+  # * "E226 missing whitespace around arithmetic operator"
+  src/pylibsshext/*.pxd: E211, E225, E226, E227, E999
+  src/pylibsshext/*.pyx: E225, E226, E227, E999
```
Comment (webknjaz) [inline .flake8:104]: These entries point to places that don't exist in the repo.

File: .flake8 [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+  WPS306  # "Found class without a base class: *" -- we have metaclass shims
+  WPS422  # "Found future import: *" -- we need these for multipython
+
+# Let's not overcomplicate the code:
+max-complexity = 10
+
+# Accessibility/large fonts and PEP8 friendly:
+#max-line-length = 79
+# Accessibility/large fonts and PEP8 unfriendly:
+max-line-length = 160
```
Comment (webknjaz) [inline .flake8:76]:
```suggestion
max-line-length = 79
```

File: .flake8 [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+
+# https://wemake-python-stylegui.de/en/latest/pages/usage/formatter.html
+format = wemake
+
+# IMPORTANT: avoid using ignore option, always use extend-ignore instead
+# Completely and unconditionally ignore the following errors:
+extend-ignore =
+  I  # flake8-isort is drunk + we have isort integrated into pre-commit
+  WPS306  # "Found class without a base class: *" -- we have metaclass shims
+  WPS422  # "Found future import: *" -- we need these for multipython
```
Comment (webknjaz) [inline .flake8:68]: not sure if this is currently needed

File: .flake8 [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+  # Cython files:
+  *.pyx
+
+# https://wemake-python-stylegui.de/en/latest/pages/usage/formatter.html
+format = wemake
+
+# IMPORTANT: avoid using ignore option, always use extend-ignore instead
+# Completely and unconditionally ignore the following errors:
+extend-ignore =
+  I  # flake8-isort is drunk + we have isort integrated into pre-commit
```
Comment (webknjaz) [inline .flake8:66]: not sure if this is necessary

File: .flake8 [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+  # PyLint configs
+  #.pylintrc
+  # tox configs
+  # tox.ini
+
+filename =
+  # Normal Python files (default):
+  *.py,
+  # Cython files:
+  *.pyx
```
Comment (webknjaz) [inline .flake8:58]: This is probably unnecessary

File: .flake8 (hunk 1 above)
Comment (slsh1o) [reply .flake8:104]: Yep, for now I just want to show errors from unrelated files.

File: .pre-commit-config.yaml [hunk 6]
```diff
@@ -61,6 +61,18 @@ repos:
...
+  rev: 3.8.3
+  hooks:
+  - id: flake8
+    language_version: python3
+    additional_dependencies:
+    - flake8-2020>=1.6.0
+    - flake8-docstrings
+    - flake8-pytest-style>=1.0.0
+    - wemake-python-styleguide
+    types: [file]
```
Comment (webknjaz) [inline .pre-commit-config.yaml:74]: this is probably unnecessary + it likely causes pre-commit to directly pass random files to flake8
```suggestion
```

File: .flake8 (hunk 1 above)
Comment (webknjaz) [reply .flake8:104]: it does literally nothing in this layout anyway

File: .pre-commit-config.yaml (hunk 6 above)
Comment (slsh1o) [reply .pre-commit-config.yaml:74]: Yeah, it is. Thx.

**Comment by slsh1o** [comment]
Ok, without wemake-python-styleguide we have no errors. Now I will add wemake plugin and then fix all errors.

**Review by webknjaz — APPROVED** [review]

**Pull request merged**